crest/Assets/Crest/Crest/Shaders/OceanHelpersDriven.hlsl

```diff
@@ -35,4 +35,20 @@ half4 SampleOceanDataAtWorldPosition(in Texture2DArray i_oceanData, in const flo
 	return result;
 }
 
+// Used to get the world position of the ocean surface from the world position by using fixed-point iteration
+float3 SampleOceanDataAtWorldPosition(in const Texture2DArray i_oceanData, in const float3 i_positionWS, in const uint i_iterations)
+{
+	const float3 queryPositionXZ = i_positionWS.xyz;
+
+	float3 undisplacedPosition = queryPositionXZ;
+	for (uint i = 0; i < i_iterations; i++)
+	{
+		const float3 displacement = SampleOceanDataAtWorldPosition(i_oceanData, undisplacedPosition);
```

Comment:
I think this may still do unnecessary work = a lod alpha will be computed (in SampleOceanDataAtWorldPosition), and lods being interpolated. I was hoping instead of calling this function, it would simply sample the lod slice directly, so this line would become:

`const float3 displacement = i_oceanData.SampleLevel( ... , WorldToUV(...));`

thinking about it, should we separate the iteration out, for example make a helper function called `InvertDisplacement(float3 displacedPosition)` or such which is specific to displacements and returns the undisplaced position given a displaced position? just wondering as this operation might be reusable elsewhere.

sorry to send you around the houses instead of giving you more specific instruction in the first place!

Reply:
if the above does work out, i'll be curious if it can be moved to *OceanHelpers.hlsl* , rather than the *Driven* version, by passing in the slice index and any other driven data as arguments. Only because then its dependencies are straightforward and clear and it can be used outside of 'driven' cases.

Reply:
So if I change to that line I won't have to sample two scales/textures?

Reply:
Yeah it should in theory just stay within one lod.

i'm waiting to see what happens at the boundaries, which could displace off-texture. i *think* it won't cause issues but we'll see how it pans out. in the worst case we'll need to revert it back to how you have it now, but i think its worth trying this simplification to avoid sampling two textures for every iteration.

Reply:
I pushed the change. Let me know if I interpreted you correctly.

Reply:
Yeah looks great! My only remaining doubt is the two functions with the same name `SampleOceanDataAtWorldPosition` differing only in args. Trying to think if we can differentiate them. `SampleOceanDataDisplacedToWorldPosition` maybe?

Reply:
Sure. Done (hopefully renamed, the right one)

Reply:
looks good nice one